### ISSUE TITLE
Support the v2 upload format

### ISSUE THREAD

**Issue by adriancole** [GoogleCloudPlatform/stackdriver-zipkin#42]
Zipkin's v2 format is a lot easier to parse (eventhough in reality it will take a while for instrumentation to start using it). It might be worthwhile supporting: openzipkin/zipkin#1684

**Comment by jsw** [comment]
I'm attempting to send v2 spans to the docker image documented here https://cloud.google.com/trace/docs/zipkin#option_1_using_a_container_image_to_set_up_your_server and getting `Malformed reading List<Span> from json`. Is this the relevant issue?

**Comment by adriancole** [comment]
Those instructions are out of date. Can you try openzipkin/zipkin-gcp image
from docker-zipkin-gcp repo or what is mentioned here?

**Comment by jsw** [comment]
Thanks for the quick response! Some combination of switching to that image and fixing my `curl` parameters gives me a 202. Where's the best place to get further troubleshooting as to why I'm not seeing the traces show up in Stackdriver Trace? I'm running the image inside k8s via `kubectl run`.

**Comment by adriancole** [comment]
if you get a 202, then there was no problem parsing the json. You can turn on
debug
<https://github.com/openzipkin/zipkin/blob/master/zipkin-server/README.md#logging>
to see if there are any upload errors to stackdriver. You can also look at
stats
<https://github.com/openzipkin/zipkin/blob/master/zipkin-server/README.md#collector>
to see if you have any drop metrics. Stats, you can either use normal
endpoint or something like our prometheus example.
<https://github.com/openzipkin/docker-zipkin#prometheus>

ex JAVA_OPTS="-Dlogging.level.zipkin=DEBUG -Dlogging.level.zipkin2=DEBUG"
enables debug

Finally, if you can use a normal zipkin server temporarily, capture a trace
and then send the v2 json back we can have a look. Ex
http://host:9411/api/v2/trace/your_trace_id

**Comment by jsw** [comment]
@adriancole Thanks for the tips. I just got it working. Some feedback that may be helpful for clearing up the documentation here and on [google](https://cloud.google.com/trace/docs/zipkin#option_1_using_a_container_image_to_set_up_your_server).

* `https://www.googleapis.com/auth/trace.append` oauth scope is required
* `gcr.io/stackdriver-trace-docker/zipkin-collector` works OOTB
* `openzipkin/zipkin-gcp` only works if I pass `STORAGE_TYPE=stackdriver` (expected?)

For reference, my commands to get each to work look like

```
kubectl run zipkin-collector --image=gcr.io/stackdriver-trace-docker/zipkin-collector --expose --port=9411
```

```
kubectl run zipkin-gcp --image=openzipkin/zipkin-gcp --env="STORAGE_TYPE=stackdriver" --expose --port=9411
```

**Comment by adriancole** [comment]
Thanks for the notes. Yes `STORAGE_TYPE=stackdriver` is required as
literally this is an extension of the normal server. For example, you can
still use cassandra storage if you want *plus* any current and future
google cloud service options. I'll try to make this more clear on the
readme. Thanks tons for the feedback and congrats getting things working!

**Comment by jsw** [comment]
Can you clarify the original intent of this issue, as it seems that v2 is supported by way of `/api/v2/spans`?

**Comment by adriancole** [comment]
Yep. We wandered a bit from that, but it is ok ;)

**Comment by jsw** [comment]
Now I'm really curious what `Support the v2 upload format` refers to, especially if it's something I might run into :-)

**Comment by adriancole** [comment]
oh. well this is done by virtue of the project re-org, as the default
server build supports v2 format. v2 format is that swagger api you
mentioned, and what all the new instrumentation use:
https://github.com/openzipkin/zipkin-reporter-java

The "v1" format still exists also via translation

**Comment by artemyarulin** [comment]
Just tried `image: gcr.io/stackdriver-trace-docker/zipkin-collector` with GCP and Kubernetes and it didn't work. Replaced it with `image: openzipkin/zipkin-gcp` with env variable `STORAGE_TYPE=stackdriver` and immediately started to see traces in Stackdriver

**Comment by adriancole** [comment]
@artemyarulin the image you mention is the right one from https://github.com/openzipkin/docker-zipkin-gcp

Is there a doc still mentioning gcr.io/stackdriver-trace-docker/zipkin-collector?

**Comment by artemyarulin** [comment]
Yeah, reading https://cloud.google.com/trace/docs/zipkin#container-image-option it says `docker pull gcr.io/stackdriver-trace-docker/zipkin-collector` should be used

**Comment by adriancole** [comment]
>
> Yeah, reading
> https://cloud.google.com/trace/docs/zipkin#container-image-option it says docker
> pull gcr.io/stackdriver-trace-docker/zipkin-collector should be used
>
@mtwo I think we meant to change this a while back.. probably forgot to
close the loop. can you help?

>

**Comment by mtwo** [comment]
Shoot, thanks for the heads up. Have the setup steps changed substantially or can I just point to the new link and leave the docs unchanged? Alternatively, should I just have that page direct people to the setup steps on GitHub?

**Comment by adriancole** [comment]
the non docker instructions change a bit as it is a layer on the
normal server https://github.com/openzipkin/zipkin-gcp/tree/master/autoconfigure/storage-stackdriver#quick-start

**Comment by adriancole** [comment]
just started for me.. of course I made a mistake and literally used the
project your_project!

will add an issue, but yeah it works. I tried a trace from netflix

**Comment by toredash** [comment]
> Ok, I've written a draft of the updated instructions. From looking at the setup steps, docker-zipkin-gcp container will pull in service account credentials automatically on GCP with the following commands:
> 
> ```
> $ docker run -d -p 9411:9411 \
> -e STORAGE_TYPE=stackdriver \
> -e GOOGLE_APPLICATION_CREDENTIALS=/root/.gcp/credentials.json \
> -e STACKDRIVER_PROJECT_ID=your_project \
> -v $HOME/.gcp:/root/.gcp:ro \
> openzipkin/zipkin-gcp
> ```
> 
> Can you confirm?

Is this still valid ? I'm not able to get this working on GKE.
